unstructured_ingest/v2/processes/connectors/couchbase.py

```diff
@@ -0,0 +1,157 @@
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    UploadContent,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+)
+
+if TYPE_CHECKING:
+    from couchbase.cluster import Cluster
+
+CONNECTOR_TYPE = "couchbase"
+SERVER_API_VERSION = "1"
+
+
+@dataclass
+class CouchbaseAccessConfig(AccessConfig):
+    connection_string: str = enhanced_field(sensitive=True)
+    username: str = enhanced_field(sensitive=True)
+    password: str = enhanced_field(sensitive=True)
+
+
+@dataclass
+class CouchbaseConnectionConfig(ConnectionConfig):
+    access_config: CouchbaseAccessConfig = enhanced_field(
+        sensitive=True, default_factory=CouchbaseAccessConfig
+    )
+    bucket: Optional[str] = None
+    scope: Optional[str] = None
+    collection: Optional[str] = None
+    batch_size: int = 50
+    connector_type: str = CONNECTOR_TYPE
+
+
+@dataclass
+class CouchbaseUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class CouchbaseUploadStager(UploadStager):
+    upload_stager_config: CouchbaseUploadStagerConfig = field(
+        default_factory=lambda: CouchbaseUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(elements_contents, output_file)
+        return output_path
+
+
+@dataclass
+class CouchbaseUploaderConfig(UploaderConfig):
+    batch_size: int = 50
+
+
+@dataclass
+class CouchbaseUploader(Uploader):
+    upload_config: CouchbaseUploaderConfig
+    connection_config: CouchbaseConnectionConfig
+    cluster: Optional["Cluster"] = field(init=False, default=None)
+    connector_type: str = CONNECTOR_TYPE
+
+    def __post_init__(self):
+        self.cluster = self.connect_to_couchbase()
+
+    @requires_dependencies(["couchbase"], extras="couchbase")
+    def connect_to_couchbase(self) -> "Cluster":
+        from datetime import timedelta
+
+        from couchbase.auth import PasswordAuthenticator
+        from couchbase.cluster import Cluster
+        from couchbase.options import ClusterOptions
+
+        connection_string = username = password = None
+        access_conf = self.connection_config.access_config
+        try:
+            if access_conf.connection_string is not None:
```

Comment:
When would this break and lead to the Exception? Also would make sense to have this as a method on the connection config class.

Reply:
Removed unused exception. Instead added a __post_init__